### PR TITLE
Refactor to ros2

### PR DESCRIPTION
**Pull request by tomdewinter94**

### CODE REVIEW

File: fadecandy_driver/CMakeLists.txt [hunk 1]
```diff
@@ -55,9 +58,19 @@ target_link_libraries(fadecandy_node ${PROJECT_NAME}_ros)
...
 )
+
+if(BUILD_TESTING)
+  install(
+    PROGRAMS test/example_client
+    DESTINATION lib/${PROJECT_NAME}
+  )
+endif()
+
+ament_package()
```
Comment (YannickdeHoop) [inline fadecandy_driver/CMakeLists.txt]: new line

File: fadecandy_driver/README.md [hunk 2]
```diff
@@ -27,7 +27,7 @@ repo. All you should need is a fadecandy board with fadecandy firmware on it.
 Start the node. Depending on permissions, this may need to run as root to talk to the USB device.
 
 ```
-rosrun fadecandy_driver fadecandy_node
+ros2 run fadecandy_driver fadecandy_driver_node
```
Comment (YannickdeHoop) [inline fadecandy_driver/README.md]: We wouldn't change the name of the executable

File: fadecandy_driver/src/ros/fadecandy_driver_ros.cpp [hunk 3]
```diff
@@ -32,28 +32,32 @@
  * POSSIBILITY OF SUCH DAMAGE.
  */
 
-#include <fadecandy_msgs/LEDArray.h>
-
 #include "./fadecandy_driver_ros.h"
 
+#include <rclcpp/logging.hpp>
+
+using std::placeholders::_1;
```
Comment (YannickdeHoop) [inline fadecandy_driver/src/ros/fadecandy_driver_ros.cpp]: why this one?

File: fadecandy_driver/test/example_client [hunk 4]
```diff
@@ -1,23 +1,27 @@
...
 
 
-def main():
-    rospy.init_node('fadecandy_client', anonymous=True)
+def main(args=None):
+    rclpy.init(args=args)
+    node_ = Node('fadecandy_client')
 
-    led_array_pub = rospy.Publisher('set_leds', LEDArray, queue_size=1)
+    led_array_pub = node_.create_publisher(LEDArray, 'set_leds', 10)
```
Comment (YannickdeHoop) [inline fadecandy_driver/test/example_client]: why 10?

Comment (YannickdeHoop) [reply fadecandy_driver/test/example_client]: ??

**Comment by YannickdeHoop** [comment]
@tomdewinter94, You may test and compile this on a robot so you make sure it works under Foxy. I guess you tested it under humble only?

File: fadecandy_driver/test/example_client (hunk 4 above)
Comment (tomdewinter94) [reply fadecandy_driver/test/example_client]: This was changed in my previous commit

**Comment by tomdewinter94** [comment]
Tested under foxy

**Pull request merged**